react-forms/react-forms/src/UserForm.tsx

```diff
@@ -0,0 +1,36 @@
+import { FormEvent, useState } from 'react';
+
+type User = {
+  username: string;
+  password: string;
+};
+
+type Props = {
+  user?: User;
+};
+
+export function UserForm({ user }: Props) {
+  const [username, setUsername] = useState(user?.username ?? '');
+  const [password, setPassword] = useState(user?.password ?? '');
+  const isCreating = user === undefined;
+
+  function handleSubmit(event: FormEvent) {
+    event.preventDefault();
+    user = { username, password };
```

Comment:
Another good way to do this, which works with the case where `user` might have fields other than `username` and `password` (for example, `userId`) is:
`user = { ...user, username, password };`